src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class PatternMatchingTests5 : PatternMatchingTestBase
+    {
+        [Fact]
+        public void ExtendedPropertyPatterns_01()
+        {
+            var program = @"
+using System;
+class C
+{
+    public C Prop1 { get; set; }
+    public C Prop2 { get; set; }
+    public C Prop3 { get; set; }
+    
+    static bool Test1(C o) => o is { Prop1.Prop2.Prop3: null };
+    static bool Test2(S o) => o is { Prop1.Prop2.Prop3: null };
+
+    public static void Main()
+    {        
+        Console.Write(Test1(new() { Prop1 = new() { Prop2 = new() { Prop3 = null }}}));
+        Console.Write(Test2(new() { Prop1 = new() { Prop2 = new() { Prop3 = null }}}));
```

Comment:
Please include calls to `Test1()` and `Test2()` where property values are `null`.